src/client/ServiceBusClient.ts

```diff
@@ -388,14 +388,30 @@ export default class ServiceBusClient implements IServiceBusClient {
         }
     }
 
-    public async purgeMessages(topic: string | null, subscription: string | null, queue: string | null): Promise<void> {
+    public async purgeMessages(
+        topic: string | null, 
+        subscription: string | null, 
+        queue: string | null, 
+        deadLetter: boolean = false): Promise<void> {
         const client = SBC.ServiceBusClient.createFromConnectionString(this.connectionString)
-        var purgeClient = null
+        var purgeClient: SBC.SubscriptionClient | SBC.QueueClient | null = null
 
-        if (topic && subscription) {
-            purgeClient = client.createSubscriptionClient(topic, subscription)
-        } else if (queue) {
-            purgeClient = client.createQueueClient(queue)
+        if (queue) {
+            if (deadLetter) {
+                let deadLetterQueueName = SBC.QueueClient.getDeadLetterQueuePath(queue)
+                purgeClient = client.createQueueClient(deadLetterQueueName)
+            } else {
+                purgeClient = client.createQueueClient(queue)
+            }
+        }
+
+        if (subscription && topic) {

```

Comment:
Actually, the client creation logic is directly copied from the `getMessages` method.
I did consider refactoring quite a bit to prevent duplication, but decided to try to fit in with the current style.

If you feel strongly about this, I would suggest that I update both places, so at least it's the same both places.

Reply:
That's fair! I totally missed the other one. In this case else if should be used since it should never go into both. Just for readability purposes. 
Thank you!